cmd/snap-confine/snap-confine.c

```diff
@@ -472,6 +472,14 @@ int main(int argc, char **argv)
 	// For classic and confined snaps
 	sc_selinux_set_snap_execcon();
 #endif
+        // Distributions using libhybris might want to have a say in the environment
+        // variables preparing hybris support correctly. If GL is not permitted the
+        // variables won't be of much use, but still set them up here.
+        if (!invocation.classic_confinement && access("/system/build.prop", F_OK) == 0) {
+            setenv("HYBRIS_LINKER_DIR", "/var/lib/snapd/lib/gl/libhybris/linker", 1);
+            setenv("HYBRIS_EGLPLATFORM_DIR", "/var/lib/snapd/lib/gl/libhybris", 1);
```

Comment:
IMO having snap-confine set up the environment by itself allows for:

- Expected environment for any Snap
- Less ways of the Snap screwing up something in the setup (because it misses -desktop-integration or does things itself)
- Have existing Snaps working as-is.

I cannot expect upstream app developers to care or have the knowledge to setup libhybris appropriately, which is why I put it there.

Reply:
Hm I think I'd rather place it somewhere around here: https://github.com/snapcore/snapd/blob/68fe7eb9f7496ec54f2b3373a5c8d1993b9eb822/cmd/snap-exec/main.go#L211-L213 there's a precedent for it already for Cups

cc @zyga ?

Reply:
Taken care of.